Make torch.cuda.gds APIs public

mikaylagawarecki opened the pull request:
Follow up to https://github.com/pytorch/pytorch/pull/145748 that turned USE_CUFILE on for CUDA 12.6 and 12.8 binaries

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #147120

mikaylagawarecki committed:
Make torch.cuda.gds APIs public

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Make torch.cuda.gds APIs public"

Follow up to https://github.com/pytorch/pytorch/pull/145748




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Make torch.cuda.gds APIs public"

Follow up to https://github.com/pytorch/pytorch/pull/145748




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Make torch.cuda.gds APIs public"

Follow up to https://github.com/pytorch/pytorch/pull/145748




[ghstack-poisoned]

albanD commented on docs/source/cuda.rst:
```diff
@@ -195,6 +195,28 @@ See the :doc:`documentation <cuda._sanitizer>` for information on how to use it.
+    gds_register_buffer
+    gds_deregister_buffer
+    GdsFile
+
```
Can you add here or link from here to an example on how to use these APIs?

albanD reviewed (commented):
Sounds good, only small nits

mikaylagawarecki commented on torch/cuda/gds.py:
```diff
@@ -31,7 +35,7 @@ def fn(*args, **kwargs):  # type: ignore[no-untyped-def]
 
 
-def _gds_register_buffer(s: Storage) -> None:
+def gds_register_buffer(s: Storage) -> None:
```
0 hits on github from repos that are not just copy pastes of pytorch so I feel ok about breaking the BC here

albanD replied:
Ok!

mikaylagawarecki committed:
Update on "Make torch.cuda.gds APIs public"

Follow up to https://github.com/pytorch/pytorch/pull/145748 that turned USE_CUFILE on for CUDA 12.6 and 12.8 binaries




[ghstack-poisoned]

albanD reviewed (approved):
Moving API to public sounds good.
Let's plan on having a tutorial (including serialization config needs) before 2.7 as an E2E example.

mikaylagawarecki committed:
Update on "Make torch.cuda.gds APIs public"

Follow up to https://github.com/pytorch/pytorch/pull/145748 that turned USE_CUFILE on for CUDA 12.6 and 12.8 binaries




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Make torch.cuda.gds APIs public"

Follow up to https://github.com/pytorch/pytorch/pull/145748 that turned USE_CUFILE on for CUDA 12.6 and 12.8 binaries




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Make torch.cuda.gds APIs public"

Follow up to https://github.com/pytorch/pytorch/pull/145748 that turned USE_CUFILE on for CUDA 12.6 and 12.8 binaries




[ghstack-poisoned]

mikaylagawarecki commented:
@pytorchbot merge
